[JavaScript] Add Array Loops concept exercise (#2126)
exercism/v3: languages/javascript/concepts/array-loops/about.md

In JavaScript, looping over `Array`s is extremely common, seeing as they are some of the most widely-used and efficient structures to store data.

> However, there are some lesser-known methods and data structures in JavaScript that handle some of the more common use cases for array looping in a more idiomatic way, like `Set`s, `Array.prototype.some`, and `Array.prototype.every`, and some immutable state prototypal functions.

Reviewer: ```suggestion
Looping over an array doesn't need to be accomplished using a `for` loop. JavaScript provides various methods that handle some of the more common use cases for array looping in a more idiomatic way, such as `Set`s, `Array.prototype.some`, and `Array.prototype.every`, as well as some immutable state prototypal functions.
```

> _JavaScript Sets_

Reviewer: ```suggestion
## JavaScript Sets
```

> Just like in mathematics, a `Set` is a collection of unique values. Typecasting to a Set from an Array is much, much easier than looping over the entire Array:

Reviewer: ```suggestion
Like in mathematics, a `Set` is a collection of unique values. Typecasting to a Set to get a list of unique items is possible:
```

```js
const numbers = [1, 1, 2, 3, 4, 5, 5, 7]

> // We need to write a function that changes the above array into an array of unique values (i.e. removes duplicates)

Reviewer: ```suggestion
// Goal: a function that changes the above array into an array of unique values (i.e. removes duplicates)
```

// This is a function that uses a for .. of loop
const makeNumbersUniqueLoop = (arr) => {
  var temp = []

>   for (const v of arr) if (arr.indexOf(v) === -1 && v !== '') temp.push(v)

Reviewer: ```suggestion
  for (const v of arr) {
    if (arr.indexOf(v) === -1 && v !== '') {
      temp.push(v)
    }
  }
```

  return temp
}

> 

Reviewer: ```suggestion

// This is a function that uses array-transformations
const makeNumbersUniqueTransform = (arr) => {
  return arr.filter((item, index, self) => self.indexOf(item) === index))
}
```

// The equivalent function using Sets and spread syntax
const makeNumbersUniqueSet = (a) => [...new Set(a)]
```

> As you can see, using different data structures can often lead to cleaner (and in most cases, faster) code.

Reviewer: ```suggestion
Using a different data structures can lead to cleaner and _intentional_ code. The upside of using a `Set` is that new elements are also considered for uniqueness, unlike the one-off loop variants.
```

> _For .. Of and Iterators_

Reviewer: ```suggestion
## `for .. of` and Iterators
```

> Although we introduced `for .. of` as a method for iterating over an `Array`, it can also be used for other classes that extend the `Iterator` class, like generator functions, `Set`s, and so on.

Reviewer: ```suggestion
Although `for .. of` has been introduced as a method for iterating over an `Array`, it can also be used for other classes that extend the `Iterator` class, like generator functions, `Set`s, and so on.
```

> _Interesting Links_

Reviewer: ```suggestion
## Interesting Links
```
